apps/site/assets/ts/shuttles/components/ShuttlesPage.tsx

```diff
@@ -1,21 +1,110 @@
-import ShuttlesOverview from "./ShuttlesOverview";
+import React, { ReactElement, useState, useEffect } from "react";
+import ShuttlesMap from "./ShuttlesMap";
+import StationDropdown from "./StationDropdown";
 import { Route } from "../../__v3api";
+import { Shape, ShuttleStop } from "./__shuttles";
 
 interface Props {
   route: Route;
+  diversions: {
+    shapes: Shape[];
+    stops: ShuttleStop[];
+  };
 }
 
-const ShuttlesPage = ({ route }: Props): ReactElement<HTMLElement> => {
+const ShuttlesPage = ({
+  route,
+  diversions
+}: Props): ReactElement<HTMLElement> => {
   const places =
     route.id === "Green" ? route.direction_names : route.direction_destinations;
 
+  const [selectedDirectionId, setSelectedDirectionId] = useState("");
+  const [displayedShapes, setDisplayedShapes] = useState([
+    ...diversions.shapes
+  ]);
+  const [displayedStops, setDisplayedStops] = useState([...diversions.stops]);
+  const [mapCenter, setMapCenter] = useState<{
+    latitude: number | null;
+    longitude: number | null;
+  }>({ latitude: null, longitude: null });
+
+  useEffect(
+    () => {
+      if (selectedDirectionId === "") {
+        setDisplayedShapes([...diversions.shapes]);
+        setDisplayedStops([...diversions.stops]);
+      } else {
+        setDisplayedShapes(
+          diversions.shapes.filter(s => s.direction_id === +selectedDirectionId)
+        );
+        setDisplayedStops(
+          diversions.stops.filter(
+            s =>
+              s.direction_id === +selectedDirectionId || s.direction_id === null
+          )
+        );
+      }
+    },
+    [diversions, selectedDirectionId]
+  );
+
+  const DirectionButton = (
+    directionId: string,
+    headsign: string
+  ): JSX.Element => (
+    <button
+      key={`${headsign}`}
+      type="button"
+      className={`btn btn-secondary btn-sm ${
+        selectedDirectionId === directionId
+          ? "c-btn-group-stackable__btn c-btn-group-stackable__btn--active"
+          : "c-btn-group-stackable__btn"
+      }`}
+      onClick={() => setSelectedDirectionId(directionId)}
+    >
+      {headsign}
+    </button>
+  );
+
   return (
     <div className="shuttles__main">
       <h2>Shuttle Maps</h2>
-      <ShuttlesOverview places={places} />
-      <div>Station Detail and dropdown</div>
-      <div>Map2</div>
+
+      <div
+        className="c-btn-group-stackable"
+        role="group"
+        aria-label="Filter shuttle map by direction"
+      >
+        {DirectionButton("", "All Directions")}
+        {Object.entries(places)
+          .sort()
+          .map(([directionId, headsign]) =>
+            DirectionButton(directionId, headsign)
+          )}
+      </div>
+
+      <ShuttlesMap shapes={displayedShapes} stops={displayedStops} />
+
+      <div className="c-heading-with-control">
+        <h3>Station Detail</h3>
+
+        <StationDropdown
+          stations={diversions.stops.filter(s => s.type === "rail_affected")}
+          onSelect={stationId => {
```

Comment:
@thecristen as a rule I like to avoid any significant amount of code in an inline event callback, could you please extract this into a method called `centerMapOnStation` or whatnot?

Reply:
Sounds good @phildarnowsky , I've gone ahead and extracted this into a method.